hw/top_earlgrey/syn/asic.constraints.sdc

```diff
@@ -16,48 +16,42 @@ puts "Applying constraints for top level"
 #####################
 # main clock        #
 #####################
-set MAIN_CLK_PIN ast_wrapper/i_ast/i_prim_clock_buf_sys/*/u_size_only_buf/${DRIVING_CELL_PIN}
+set MAIN_CLK_PIN ast_wrapper/i_ast/i_sys_clk/i_sys_osc/sys_clk_o
 set MAIN_RST_PIN IO_RST_N
-# target is 100MHz, overconstrain to 125 MHz (+25%)
-set MAIN_TCK  8.0
-set_ideal_network [get_pins ${MAIN_CLK_PIN}]
-set_ideal_network [get_ports ${MAIN_RST_PIN}]
-
-create_clock -name MAIN_CLK -period ${MAIN_TCK} [get_pins ${MAIN_CLK_PIN}]
+# target is 100MHz, overconstrain by factor 
+set MAIN_TCK_TARGET_PERIOD  10
+set MAIN_TCK_PERIOD [expr $MAIN_TCK_TARGET_PERIOD*$FACTOR] ;# over constraining 
+# For now we remove this as clock is, by default, ideal. Reset, we'll try w/o ideal_network.
+#set_ideal_network [get_pins ${MAIN_CLK_PIN}]
+#set_ideal_network [get_ports ${MAIN_RST_PIN}]
+
+create_clock -name MAIN_CLK -period ${MAIN_TCK_PERIOD} [get_pins ${MAIN_CLK_PIN}]
 set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks MAIN_CLK]
 
 #####################
 # USB clock         #
 #####################
-set USB_CLK_PIN ast_wrapper/i_ast/i_prim_clock_buf_usb/*/u_size_only_buf/${DRIVING_CELL_PIN}
-# 50MHz
-set USB_TCK 20.0
-set_ideal_network [get_pins ${USB_CLK_PIN}]
+set USB_CLK_PIN ast_wrapper/i_ast/i_usb_clk/i_usb_osc/usb_clk_o
+# target is 48MHz, overconstrain by 5% 
+set USB_TCK_TARGET_PERIOD 20.8
+set USB_TCK_PERIOD [expr $USB_TCK_TARGET_PERIOD*$FACTOR]
+#set_ideal_network [get_pins ${USB_CLK_PIN}]
 
-create_clock -name USB_CLK -period ${USB_TCK} [get_pins ${USB_CLK_PIN}]
+create_clock -name USB_CLK -period ${USB_TCK_PERIOD} [get_pins ${USB_CLK_PIN}]
 set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks USB_CLK]
 
-set USB_IN_DEL_FRACTION 0.55
-set USB_OUT_DEL_FRACTION 0.55
-set USB_IN_DEL    [expr ${USB_IN_DEL_FRACTION} * ${USB_TCK}]
-set USB_OUT_DEL   [expr ${USB_OUT_DEL_FRACTION} * ${USB_TCK}]
-
-# constrain dedicated USB IOs
-set_input_delay ${USB_IN_DEL} [get_ports USB_P] -clock USB_CLK
-set_input_delay ${USB_IN_DEL} [get_ports USB_N] -clock USB_CLK
-
-set_output_delay ${USB_OUT_DEL} [get_ports USB_P] -clock USB_CLK
-set_output_delay ${USB_OUT_DEL} [get_ports USB_N] -clock USB_CLK
-
+# TBD - add 1ns max_delay constraint from pin to differential receiver and from usb_d_o 
+# AI (Google) - send path to driver and receiver 
```

Comment:
differential input paths:
```
[get_ports USB_N] -> [get_pinsget_pins top_earlgrey/u_usbdev_aon/i_usbdev_iomux/cdc_io_to_usb/gen_*u_impl_*/u_sync_1/gen_*u_impl*/gen_flops_2*_u_size_only_reg/D]
[get_ports USB_P] -> [top_earlgrey/u_usbdev_aon/i_usbdev_iomux/cdc_io_to_usb/gen_*u_impl_*/u_sync_1/gen_*u_impl*/gen_flops_3*_u_size_only_reg/D]
]
```
single ended input paths go through diff input buffer:
```
[get_ports USB_*] -> [get_pins top_earlgrey/u_usbdev_aon/i_usbdev_iomux/cdc_io_to_usb/gen_*u_impl_*/u_sync_1/gen_*u_impl*/gen_flops_1*_u_size_only_reg/D]
]
```

output paths:
```
[get_pins top_earlgrey/u_usbdev_aon/usbdev_impl/u_usb_fs_nb_pe/u_usb_fs_tx/usb_d_q_reg/Q] -> [get_ports USB_*]
[get_pins top_earlgrey/u_usbdev_aon/usbdev_impl/u_usb_fs_nb_pe/u_usb_fs_tx/oe_q_reg/Q] -> [get_ports USB_*]
```

Reply:
> 
> 
> Thanks @arnonsha! This LGTM in general.
> Also, there are two empty merge commits in this PR that seem unnecessary. Can you remove those?

I think there's only one related to the update I did for my fork.
By removing do you mean "squshing" ?

Reply:
so in general, we try to always cleanly rebase our work such that we do not get any of these merge commits. that results in a much cleaner, linear history (for instance, use `git pull --rebase` instead of `git pull`). 

in this particular case, do you have any other local, uncommited changes in your repository?
if not, you can cherry-pick the SDC change on top of a new and fresh checkout of `v1-bronze` in order to get rid of these merge commits:

```shell
# make sure you are on this branch
git checkout sdc_update

# backup this branch
git checkout -b sdc_update_bak

# go back and fetch upstream
git checkout sdc_update
git fetch upstream

# make sure you are on sdc_update before executing the following reset command
# this resets the state of sdc_update to v1-bronze
git reset --hard upstream/v1-bronze

# cherry pick the SDC update commit onto this branch
git cherry-pick da82ff066a00bc24fa79ab474cb76d382356dadc

# push the update to this PR
git push -f origin sdc_update
```

Reply:
OK, updated